pkg/controllers/cassandra/service/seedprovider/resource.go

```diff
@@ -27,6 +27,10 @@ func updateServiceForCluster(
 	service.SetName(util.SeedProviderServiceName(cluster))
 	service.Spec.Type = apiv1.ServiceTypeClusterIP
 	service.Spec.ClusterIP = "None"
+
+	// Only mark nodes explicitly labeled as seeds as seed nodes
+	service.Spec.Selector["seed"] = "true"
+
```

Comment:
:grey_question: Sorry about my over complicated code here :-( I was trying to share as much between the two services as possible, hence the `SetStandardServiceAttributes` stuff. 
But now I'm thinking that the CQL service probably isn't necessary and all this could be simplified in a future branch.